sidecar/src/self_telemetry.rs

```diff
@@ -64,6 +69,42 @@ impl<'a> MetricData<'a> {
             ));
         }
 
+        if trace_metrics.api_requests > 0 {
+            futures.push(self.send(
+                self.trace_api_requests,
+                trace_metrics.api_requests as f64,
+                vec![],
+            ));
+        }
+        if trace_metrics.api_errors_network > 0 {
+            futures.push(self.send(
+                self.trace_api_errors,
+                trace_metrics.api_errors_network as f64,
+                vec![Tag::new("type", "network").unwrap()],
+            ));
+        }
+        if trace_metrics.api_errors_timeout > 0 {
+            futures.push(self.send(
+                self.trace_api_errors,
+                trace_metrics.api_errors_timeout as f64,
+                vec![Tag::new("type", "timeout").unwrap()],
+            ));
+        }
+        if trace_metrics.api_errors_status_code > 0 {
+            futures.push(self.send(
+                self.trace_api_errors,
+                trace_metrics.api_errors_status_code as f64,
+                vec![Tag::new("type", "status_code").unwrap()],
+            ));
+        }
+        for (status_code, count) in &trace_metrics.api_responses_count_per_code {
+            futures.push(self.send(
+                self.trace_api_responses,
+                *count as f64,
+                vec![Tag::new("status_code", status_code.to_string().as_str()).unwrap()],
```

Comment:
The most minor of nitpicks, feel free to ignore: `&status_code.to_string()` also works and is a bit less verbose.